src/main/jflex/VentoLexer.flex

```diff
@@ -57,68 +61,78 @@ EMPTY_LINE=(\r\n|\r|\n)[ \t]*(\r\n|\r|\n)
 
 <YYINITIAL> {
 
+    {EMPTY_LINE}              { return VentoLexerTypes.EMPTY_LINE; }
+    {WHITESPACE}              { return com.intellij.psi.TokenType.WHITE_SPACE; }
+    {HTML_TAG}                { return VentoLexerTypes.HTML_TAG; }
+    {TEXT}                    { return VentoLexerTypes.TEXT; }
 
+    {OPEN_COMMENT_TRIMMED_PHRASE}    {
+                yybegin(COMMENT);
+                return VentoLexerTypes.OPEN_TRIM_COMMENT_CLAUSE;
+            }
 
-    {EMPTY_LINE}              { return VentoTypes.EMPTY_LINE; }
-    {WHITESPACE}              { return com.intellij.psi.TokenType.WHITE_SPACE;}
-    {HTML_TAG}                { return VentoTypes.HTML_TAG; }
-    {TEXT}                    { return VentoTypes.TEXT; }
-
-    {TRIMMED_COMMENT_START}    {
-        yybegin(COMMENTED_CONTENT);
-        return VentoTypes.TRIMMED_COMMENTED_START;
-    }
-
-    {COMMENT_START}    {
-        yybegin(COMMENTED_CONTENT);
-        return VentoTypes.COMMENTED_START;
-    }
+    {OPEN_COMMENT_PHRASE}    {
+            yybegin(COMMENT);
+            return VentoLexerTypes.OPEN_COMMENT_CLAUSE;
+         }
 
     {JAVASCRIPT_START}    {
         yybegin(SCRIPT_CONTENT);
-        return VentoTypes.JAVASCRIPT_START;
+        return VentoLexerTypes.JAVASCRIPT_START;
     }
 
     {VARIABLE_START}    {
         yybegin(VARIABLE_CONTENT);
-        return VentoTypes.VARIABLE_START;
+        return VentoLexerTypes.VARIABLE_START;
     }
 
-    [^] { return VentoTypes.ERROR; }
+    [^] { return VentoLexerTypes.ERROR; }
 
 }
 
 <VARIABLE_CONTENT> {
-    \|\| {return VentoTypes.VARIABLE_PIPES;}
-    ([^}\|]|"}"[^}\|])+ { return VentoTypes.VARIABLE_ELEMENT; }
+    \|\| {return VentoParserTypes.VARIABLE_PIPES;}
+    ([^}\|]|"}"[^}\|])+ { return VentoLexerTypes.VARIABLE_ELEMENT; }
     "}}" {
-       yybegin(YYINITIAL);
-       return VentoTypes.VARIABLE_END;
-    }
+           yybegin(YYINITIAL);
+           return VentoLexerTypes.VARIABLE_END;
+        }
 }
 
 
 <SCRIPT_CONTENT> {
-   ([^}]|"}"[^}])+ { return VentoTypes.JAVASCRIPT_ELEMENT; }
+   ([^}]|"}"[^}])+ { return VentoParserTypes.JAVASCRIPT_ELEMENT; }
    "}}" {
-       yybegin(YYINITIAL);
-       return VentoTypes.JAVASCRIPT_END;
-   }
+          yybegin(YYINITIAL);
+          return VentoLexerTypes.JAVASCRIPT_END;
+      }
 }
 
-<COMMENTED_CONTENT> {
+<COMMENT> {
 
-    [^-#{]+ { return VentoTypes.COMMENTED_CONTENT; }
+    // Match everything that is not the start of a closing comment sequence
+    ([^#-]|"#"[^}]|"-"[^#}])+ { return VentoLexerTypes.COMMENTED_CONTENT; }
 
-    "#}}" {
-        yybegin(YYINITIAL);
-        return VentoTypes.COMMENTED_END;
+    // Handle single characters that might be part of closing sequences
+    "#" { return VentoLexerTypes.COMMENTED_CONTENT; }
+    "-" { return VentoLexerTypes.COMMENTED_CONTENT; }
```

Comment:
How does the plugin know what order to check these in? These two need to be checked _after_ the `CLOSE_COMMENT_PHRASE` and the trimmed version. This might already work fine, just the ordering seems a bit odd to me.